python/sdk/prelude_sdk/controllers/build_controller.py

```diff
@@ -80,3 +80,65 @@ def upload(self, test_id, filename, data):
         if res.status_code == 200:
             return res.json()
         raise Exception(res.text)
+
+    @verify_credentials
+    def create_threat(self, name, threat_id=None, source_id=None, source=None, published=None, tests=None):
+        """ Create a threat """
+        body = dict(name=name)
+        if threat_id:
+            body['id'] = threat_id
+        if source_id:
+            body['source_id'] = source_id
+        if source:
+            body['source'] = source
+        if published:
+            body['published'] = published
+        if tests:
+            body['tests'] = tests
+
+        res = requests.post(
+            f'{self.account.hq}/build/threats',
+            json=body,
+            headers=self.account.headers,
+            timeout=10
+        )
+        if res.status_code == 200:
+            return res.json()
+        raise Exception(res.text)
+
+    @verify_credentials
+    def update_threat(self, threat_id, name=None, source_id=None, source=None, published=None, tests=None):
+        """ Update a threat """
+        body = dict()
+        if name:
```

Comment:
how come this is not None?

Reply:
it depends on if the field can is nullable. so `if name` b/c name cannot be empty string / null. `if source_id is not None` b/c that can be empty